### PR TITLE
GitHub 37

### PR DESCRIPTION
**Pull request by QuantumPlumber**
Added a general function for timeseries edge case detection. 

One edge case specified so far: when window length required by the handler is greater than the supplied data length.

### CODE REVIEW

File: dataprep_ml/cleaners.py [hunk 1]
```diff
@@ -313,11 +336,13 @@ def _remove_columns(data: pd.DataFrame, identifiers: Dict[str, object], target:
     :param identifiers: A dict containing all identifier typed columns
     :param target: The target columns
     :param mode: Can be "predict" or "train"
-    :param timeseries_settings: Timeseries related settings, only relevant for timeseries predictors, otherwise can be the default object
+    :param timeseries_settings: Timeseries related settings, only relevant for timeseries predictors, otherwise can be 
+    the default object
```
Comment (paxcema) [inline dataprep_ml/cleaners.py:340]: nitpick: extra newline? Should be ignored by the linter anyway, unless this somehow improves the docs formatting (in which case, it's a welcome change!)

File: dataprep_ml/cleaners.py [hunk 2]
```diff
@@ -302,6 +304,27 @@ def _rm_rows_w_empty_targets(df: pd.DataFrame, target: str) -> pd.DataFrame:
...
+    :param timeseries_settings: Timeseries related settings, only relevant for timeseries predictors, otherwise can be
+    the default object
+
+    :returns: None
+
+    """
+
+    if timeseries_settings.get('window', None) is not None:
+        window = timeseries_settings.get('window', 0)
+        assert len(data) >= window, "Window size is greater than input data length."
```
Comment (paxcema) [inline dataprep_ml/cleaners.py]: Nit: maybe add a recommendation, e.g. "[...] Please set a smaller window size or try a larger dataset".

**Review by paxcema — APPROVED** [review]
New method certainly solves the linked issue. Perhaps a simple unit test would be good out of habit/process, but I won't block merging because its logic is quite simple.

**Comment by paxcema** [comment]
Oh, another thing @QuantumPlumber: please change the title to be more descriptive of the problem this solves (or any added changes).

Congrats on your first PR! 🎉

**Comment by QuantumPlumber** [comment]
@paxcema I'm not sure what title you refer too.. is it the title of the PR or the branch, perhaps the last commit?

**Comment by paxcema** [comment]
Yep I meant the title of the PR so that it is more descriptive than just a mention of the issue it fixes 👍

**Pull request merged**